src/dict_mem.cpp

```diff
@@ -94,11 +95,18 @@ DictMem::DictMem(const std::string &mbdir, bool init_header, size_t memsize,
             Destroy();
             throw (int) MBError::INVALID_SIZE;
         }
+        is_valid = true;
```

Comment:
is_valid should be set after all initializations are successful. What if the new operator at line 111 failed and is_valid is set to true?